tests/functional/conftest.py

```diff
@@ -48,10 +55,31 @@
 # Directory storing test-specific data.
 _DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
-# Provide test access to _DATA_DIR via a fixture.
+# Code
+# ====
+# Fixtures
+# --------
+# Define a fixure for the DataDir object.
 @pytest.fixture
-def data_dir():
-    return _DATA_DIR
+def data_dir(
+  # The request object for this test. See
+  # https://pytest.org/latest/builtin.html#_pytest.python.FixtureRequest
+  # and
+  # https://pytest.org/latest/fixture.html#fixtures-can-introspect-the-requesting-test-context.
+  request,
+  # The tmpdir object for this test. See
+  # https://pytest.org/latest/tmpdir.html.
+  tmpdir):
+
+    # Strip the leading 'test_' from the test's name.
+    name = request.function.__name__[5:]
```

Comment:
I can think of several approaches:

- For each shared data-folder, create a new fixture that calls `data_dir` with a fixed data-folder name.
- Apply a `data_dir_folder` mark to each function, whose first argument is the data-folder name. `data_dir` will use this folder name instead of using the test's name, if present.
- (If you want to also run the same test for several data-sets) Parametrize the test on the data_dir fixture.